src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/ExplainTest.java

```diff
@@ -0,0 +1,128 @@
+import com.amazon.opendistroforelasticsearch.sql.exception.SqlParseException;
+import com.amazon.opendistroforelasticsearch.sql.utils.JsonPrettyFormatter;
+import com.amazon.opendistroforelasticsearch.sql.query.ESActionFactory;
+import com.amazon.opendistroforelasticsearch.sql.query.QueryAction;
+import com.amazon.opendistroforelasticsearch.sql.query.SqlElasticRequestBuilder;
+import com.amazon.opendistroforelasticsearch.sql.request.SqlRequest;
+import com.amazon.opendistroforelasticsearch.sql.util.CheckScriptContents;
+import com.google.common.io.Files;
+import org.elasticsearch.client.Client;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLFeatureNotSupportedException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class ExplainTest {
+
+    @Test
+    public void assertExplainRequestShouldBeJson() throws IOException {
+
+        String expectedOutputFilePath = TestUtils.getResourceFilePath(
+                "src/test/resources/expectedOutput/explain_format_pretty.json");
+        String expectedOutput = Files.toString(new File(expectedOutputFilePath), StandardCharsets.UTF_8)
+                .replaceAll("\r", "");
+
+        String explainResult = explain(String.format("{\"query\":\"" +
+                "SELECT firstname " +
+                "FROM %s\"}", TestsConstants.TEST_INDEX_ACCOUNT));
+        String result = (new JsonPrettyFormatter()).format(explainResult);
+
+        assertThat(result.replaceAll("\\s", ""),
+                equalTo(expectedOutput.replaceAll("\\s", "")));
+    }
+
+    @Test
+    public void assertExplainRequestPrettyFormatted() throws IOException {
+
+        String expectedOutputFilePath = TestUtils.getResourceFilePath(
+                "src/test/resources/expectedOutput/explain_format_pretty.json");
+        String expectedOutput = Files.toString(new File(expectedOutputFilePath), StandardCharsets.UTF_8);
+        removeBlank(expectedOutput);
+        replaceReturn(expectedOutput);
+
+        String explainResult = explain(String.format("{\"query\":\"" +
+                "SELECT firstname " +
+                "FROM %s\"}", TestsConstants.TEST_INDEX_ACCOUNT));
+        String result = (new JsonPrettyFormatter()).format(explainResult);
+        removeBlank(result);
+        replaceReturn(result);
+
+        assertThat(result, equalTo(expectedOutput));
+    }
+
+    @Test
+    public void assertExplainJoinRequestPrettyFormatted() throws IOException {
+
+        String expectedOutPutFilePath = TestUtils.getResourceFilePath(
+                "src/test/resources/expectedOutput/explain_join_format_pretty.json");
+        String expectedOutput = Files.toString(new File(expectedOutPutFilePath), StandardCharsets.UTF_8);
+        removeBlank(expectedOutput);
+        replaceReturn(expectedOutput);
```

Comment:
why these post processing is needed for String compare?

Reply:
No need to post process